src/Tickets/Admin/All_Tickets/Provider.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * The main service provider for the Tickets Admin Attendees page.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Admin
+ */
+
+namespace TEC\Tickets\Admin\All_Tickets;
+
+use TEC\Common\Contracts\Service_Provider;
+
+/**
+ * Service provider for the Tickets Admin All Tickets
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Admin
+ */
+class Provider extends Service_Provider {
+
+	/**
+	 * Register the provider singletons.
+	 *
+	 * @since TBD
+	 */
+	public function register() {
+		$this->register_hooks();
+		$this->register_assets();
+
+		// Register the SP on the container.
+		$this->container->singleton( static::class, $this );
```

Comment:
providers aren't usually (never?) singletons - use register_provider.
In fact, [you do below](https://github.com/the-events-calendar/event-tickets/pull/3164/files#diff-24f4d9f7d22216dc2143f2b48b826a6d9c0dafb93f829e679023a34159273197R38) You shouldn't have to do both.